src/ruby/ext/grpc/rb_channel.c

```diff
@@ -261,6 +265,65 @@ static VALUE grpc_rb_channel_init(int argc, VALUE *argv, VALUE self) {
   return self;
 }
 
+/*
+  call-seq:
+    ch = Channel.insecure_create_from_fd("fd:42", 42, {'arg1': 'value1'})
+
+  Creates channel instances from an existing file descriptor. */
+static VALUE grpc_rb_channel_insecure_create_from_fd(
+    VALUE klass, VALUE rb_target, VALUE rb_fd, VALUE rb_args) {
+#ifdef GPR_SUPPORT_CHANNELS_FROM_FD
```

Comment:
rather than placing an `ifdef` around this larger section of code, can we please instead create a 
ruby wrapper" for `grpc_insecure_channel_create_from_fd`? e.g., `grpc_rb_insecure_channel_create_from_fd`?, which handles the platform compatiblity and throws `NotImplemented`?

the `ifdef` could then wrap a smaller chunk of code within that.

Reply:
I don't understand this comment. `grpc_rb_channel_insecure_create_from_fd` is a ruby wrapper that behaves as you describe. It's not clear how to make the ifdef smaller, since it would be wasteful to include code that can't be executed on the current platform.

Reply:
Phrased more clearly, can we move all of the code that's currently inside of the ifdef into it's own function, and then put only the function call inside the ifdef? I'd just like to reduce the number of lines inside this `ifdef`.